File: sol/solution/0001-0100/0027. Remove Element/Solution.php
class Solution {
    /**
     * @param Integer[] $nums
     * @param Integer $val
     * @return Integer
     */
    function removeElement(&$nums, $val) {
        $n = count($nums); $i = 0;
        while ($i < $n) {
            if ($nums[$i] === $val) {
                $nums[$i] = $nums[$n - 1];
                $n--;
            } else {
                $i++;
            }
        }
        return $n;
    }
}

File: sol/solution/0001-0100/0027. Remove Element/Solution.scala
object Solution {
  def removeElement(nums: Array[Int], v: Int): Int = {
    var n = nums.length
    var i = 0
    while (i < n) {
      if (nums(i) == v) {
        nums(i) = nums(n - 1)
        n -= 1
      } else {
        i += 1
      }
    }
    n
  }
}